client/src/components/R&R/ReviewsBreakdown.jsx

```diff
@@ -5,11 +5,84 @@ import { ProductContext } from '../../context';
 
 const ReviewsBreakdown = () => {
   const { reviews, reviewsMeta, reviewSort, setReviewSort } = useContext(ProductContext);
+  const options = {
+    Size: ['A size too small', 'Perfect', 'A size too big'],
+    Width: ['Too narrow', 'Perfect', 'Too wide'],
+    Comfort: ['Uncomfortable', 'Ok', 'Perfect'],
+    Quality: ['Poor', 'What I expected', 'Perfect'],
+    Length: ['Runs short', 'Perfect', 'Runs long'],
+    Fit: ['Runs tight', 'Perfect', 'Runs long'],
+  };
+
+  const recommendPercentage = () => {
+    const total = Number(reviewsMeta.recommended.true) + Number(reviewsMeta.recommended.false);
+    return Math.round((reviewsMeta.recommended.true / total) * 100);
+  };
+
+  const characteristicArray = [];
+  const formatCharacteristics = () => {
+    Object.keys(reviewsMeta.characteristics).map((key) => {
+      const obj = { ...reviewsMeta.characteristics[key] };
+      obj.char = key;
+      characteristicArray.push(obj);
+    });
+    return characteristicArray;
+  };
+
+  const starPercents = {};
+  const keys = [];
+  const ratingPercents = () => {
+    let ratingTotal = 0;
+    Object.keys(reviewsMeta.ratings).forEach((key) => {
+      keys.push(key);
+    });
+    keys.sort((a, b) => {
+      return b - a;
+    });
+    keys.forEach((key) => {
+      ratingTotal += Number(reviewsMeta.ratings[key]);
+    });
+    keys.forEach((key) => {
+      starPercents[key] = ((reviewsMeta.ratings[key] / ratingTotal) * 100).toFixed(0);
+    });
+    return keys;
+  };
```

Comment:
This function should probably be renamed to `calculateRatingPercentages` or something better than that. And shorter.